portality/static/widget/fixed_query_body_dev_edges.html

```diff
@@ -0,0 +1,68 @@
+<script type="text/javascript" src="/static/vendor/jquery-3.4.1/jquery-3.4.1.min.js"></script>
+<script type="text/javascript" src="/static/js/doaj.js"></script>
+<link rel="stylesheet" href="/static_content/_site/css/main.css">
+
+<div></div>
+<div class="facetview">
+</div>
+
+<script type="text/javascript">
+
+    $.extend(true, doaj, {
+        publicSearchConfig : {
+            publicSearchPath : '/query/journal,article/_search?ref=fqw',
+            lccTree: '{{ lcc_tree|tojson }}'
+        }
+    });
+
+    var widget_fv_opts = {};
+
+    // If options have been set using the Export method, use only those
+    if (typeof SEARCH_CONFIGURED_OPTIONS != 'undefined') {
+        widget_fv_opts["queryString"] = SEARCH_CONFIGURED_OPTIONS;
+        widget_fv_opts["rawQuery"] = true;
+    }
+    // Otherwise, use options provided.
+    else if (typeof QUERY_OPTIONS != 'undefined') {
+        widget_fv_opts["rawQuery"] = false;
+
+        if(QUERY_OPTIONS.hasOwnProperty('selector')){
+            widget_fv_opts['selector'] = QUERY_OPTIONS.selector;
+        }
+
+        widget_fv_opts[`initialParams`] = {};
+        if (QUERY_OPTIONS.hasOwnProperty('size')) {
+            widget_fv_opts['initialParams']['size'] = QUERY_OPTIONS.size;
+        }
+        if (QUERY_OPTIONS.hasOwnProperty('from')) {
+            widget_fv_opts['initialParams']['from'] = QUERY_OPTIONS.from;
+        }
+
+        widget_fv_opts[`queryString`] = {};
+        if (QUERY_OPTIONS.hasOwnProperty('query_string')) {
+            widget_fv_opts['queryString']['queryString'] = QUERY_OPTIONS.query_string;
+        }
+        if (QUERY_OPTIONS.hasOwnProperty('query_field')) {
+            widget_fv_opts['queryString']['defaultField'] = QUERY_OPTIONS.query_field;
+        }
+        if (QUERY_OPTIONS.hasOwnProperty('search_operator')) {
+            widget_fv_opts['queryString']['defaultOperator'] = QUERY_OPTIONS.search_operator;
+        }
+
+        widget_fv_opts['sortFilter'] = {}
+        if (QUERY_OPTIONS.hasOwnProperty('sort_field')) {
+            widget_fv_opts['sortFilter']['field'] = QUERY_OPTIONS.sort_field;
+        }
+        if (QUERY_OPTIONS.hasOwnProperty('sort_direction')) {
+            widget_fv_opts['sortFilter']['sort_direction'] = QUERY_OPTIONS.sort_direction;
+        }
+
+        if (QUERY_OPTIONS.hasOwnProperty('search_type')) {
+            widget_fv_opts['es_type'] = QUERY_OPTIONS.search_type;
+        }
+
+    }
```

Comment:
As much as possible of this should be in the widget js file, rather than embedded in the html file

Reply:
I think this is the code that we were discussing how to map from the options to the ES query.  Can we get rid of this whole conversion to the intermediate `widget_fv_opts` object, and just generate a query directly that we pass into the init function.  Since all these options are in a global `QUERY_OPTIONS` object, I don't think we need the `widget_fv_options` object at all, the widget init can just make the direct coversion to an ES query object.